mautic/mautic
# Pipedrive cleanUpFields issue

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Yes
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | https://github.com/mautic/mautic/issues/5325
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

This bugfix/feature has been sponsored by @Webmecanik

[//]: # ( Required: )
#### Description:
Issue https://github.com/mautic/mautic/issues/5325 si related to new way  how to handle integration with objects. This PR should fix it.
Pipedrive doesn't clean  deleted fields because cleanUpFields https://github.com/mautic/mautic/blob/00f3d5f58cdbaad5a18ad1b13be3ef248b15e62d/app/bundles/PluginBundle/Controller/PluginController.php#L230 working with $settings and $objects context. Pipedrive integration reset settings https://github.com/mautic/mautic/blob/staging/plugins/MauticCrmBundle/Integration/PipedriveIntegration.php#L152
This PR reworked this with same logic like vTiger integration

[//]: # ( As applicable: )
#### Steps to reproduce the bug:

Require setup Pipedrive with trial account https://mautic.org/docs/en/plugins/pipedrive.html 



1. Create a custom field in Pipedrive
2. Map it in integration mapping tab
3. Delete the custom field in Pipedrive
4. Notice that in the integration_settings table the mapping with the deleted field is still existing. You can try to open the mapping and save & close but in DB the value stays
5. Push a contact and notice error "the field XXX is not existing"

#### Steps to test this PR:
1.  Setup Pipedrive first, then  try run migrations: `php app/console doctrine:migrations:migrate`
We need migrations, because Pipedrive integration worked with contacts object as default. Now we add this object to features settings of plugin, but due the BC break, we need enable contacts object for older installations.
2.  Then go to plugin and see if everything work like before
3. Then repeat steps to reproduce
4: Works If fields are clean up after save settings
